Makefile

```diff
@@ -68,27 +69,22 @@ else
 all: out/Makefile $(NODE_EXE) $(NODE_G_EXE)
 endif
 
-# The .PHONY is needed to ensure that we recursively use the out/Makefile
-# to check for changes.
-.PHONY: $(NODE_EXE) $(NODE_G_EXE)
+$(NODE_EXE): out/Release/node out/Makefile
+	ln -fs $< $@
 
-# The -r/-L check stops it recreating the link if it is already in place,
-# otherwise $(NODE_EXE) being a .PHONY target means it is always re-run.
-# Without the check there is a race condition between the link being deleted
-# and recreated which can break the addons build when running test-ci
-# See comments on the build-addons target for some more info
-$(NODE_EXE): config.gypi out/Makefile
+$(NODE_G_EXE): out/Debug/node out/Makefile
+	ln -fs $< $@
+
+out/Release/node: $(ALL_DEPS)
```

Comment:
Does `$(ALL_DEPS)` include `out/Makefile`? Since this target recursively executes make it should have a dependency on the makefile.